### PR TITLE
DOCS-7906 Code Analysis Documentation Updates

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do? What is the motivation?

Update the Setup instructions for Code Analysis based on the in-app two tab setup: Datadog-hosted scans or in your CI pipeline and adds screenshots where appropriate.

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/code_analysis/_index.md
Comment (alai97) [inline content/en/code_analysis/_index.md:127]:
```suggestion
[101]: /code_analysis/static_analysis/github_actions 
[102]: /code_analysis/static_analysis/circleci_orbs 
[103]: /code_analysis/static_analysis/generic_ci_providers 
[104]: /code_analysis/software_composition_analysis/github_actions 
[105]: /code_analysis/software_composition_analysis/generic_ci_providers 
```

File: content/en/code_analysis/_index.md
Comment (drichards-87) [inline content/en/code_analysis/_index.md:96]:
```suggestion
<div class="alert alert-warning">Datadog-hosted scans are supported by Software Composition Analysis (SCA) and GitHub repositories only. To enable Static Analysis or use a different CI provider, run scans in your CI pipelines instead.</div>
```

Comment (drichards-87) [reply content/en/code_analysis/_index.md:96]: Not sure if this is the most accurate way to describe alternative repos other than Github.

**Review by drichards-87 — COMMENTED** [review]
@alai97 Left a small suggestion and approved the PR. Are you still waiting for others to review the PR before merging?

**Review by drichards-87 — APPROVED** [review]

File: content/en/code_analysis/_index.md
Comment (markazerdd) [inline content/en/code_analysis/_index.md:111]:
```suggestion
If you do not want to run your scans directly through Datadog, you can select which scans you'd like to run ([Static Analysis][106] and [Software Composition Analysis][107]) and configure your CI pipeline provider accordingly.
```

File: content/en/code_analysis/software_composition_analysis/_index.md
Comment (markazerdd) [inline content/en/code_analysis/software_composition_analysis/_index.md:38]:
```suggestion
You can run SCA scans on your repositories directly through Datadog or in your CI pipelines by using [Code Analysis][3] and get runtime monitoring capabilities by using [Datadog Application Security][1].
```

File: static/images/integrations_logos/cpp_avatar.svg
Comment (markazerdd) [inline static/images/integrations_logos/cpp_avatar.svg:1]: We support C# not C++, so we should be using the .NET logo instead

**Review by markazerdd — CHANGES_REQUESTED** [review]

File: content/en/code_analysis/_index.md
Comment (alai97) [reply content/en/code_analysis/_index.md:36]:
```suggestion
Code Analysis displays results for violations found by [Static Analysis][1] and [Software Composition Analysis (SCA)][2] scans in your repositories. 
```

File: content/en/code_analysis/_index.md
Comment (alai97) [inline content/en/code_analysis/_index.md:156]:
```suggestion
[11]: /code_analysis/ide_plugins/
[12]: /integrations/github/
```

File: layouts/partials/code_analysis/languages-getting-started.html
Comment (alai97) [inline layouts/partials/code_analysis/languages-getting-started.html:36]:
```suggestion
            {{ partial "img.html" (dict "root" . "src" "integrations_logos/dotnet_avatar.svg" "class" "img-fluid" "alt" "c sharp" "width" "200") }}
```

File: content/en/code_analysis/_index.md
Comment (markazerdd) [inline content/en/code_analysis/_index.md:100]: The hyperlink for the creating a new GitHub App isn't rendering correctly, I think the URL is supposed to be [102] instead of [12]?

File: content/en/code_analysis/_index.md
Comment (alai97) [inline content/en/code_analysis/_index.md:159]:
```suggestion
```

File: content/en/code_analysis/_index.md
Comment (alai97) [reply content/en/code_analysis/_index.md:100]:
```suggestion
To enable [Software Composition Analysis][101] on GitHub repositories, click **Select Repositories** on your desired GitHub account and click the toggle for `Enable Software Composition Analysis (SCA)` to enable for all repositories. If you don't see any GitHub accounts listed, [create a new GitHub App][102] to get started.
```

File: content/en/code_analysis/_index.md
Comment (alai97) [inline content/en/code_analysis/_index.md:108]:
```suggestion
[101]: /code_analysis/software_composition_analysis
[102]: /integrations/github/
```

**Review by markazerdd — APPROVED** [review]

**Pull request merged**
